src/main/java/com/gmail/goosius/siegewar/playeractions/DestroyBlock.java

```diff
@@ -0,0 +1,168 @@
+import com.gmail.goosius.siegewar.enums.SiegeWarPermissionNodes;
+import com.gmail.goosius.siegewar.objects.Siege;
+import com.gmail.goosius.siegewar.settings.SiegeWarSettings;
+import com.gmail.goosius.siegewar.settings.Translation;
+import com.gmail.goosius.siegewar.utils.SiegeWarAllegianceUtil;
+import com.gmail.goosius.siegewar.utils.SiegeWarBlockUtil;
+import com.gmail.goosius.siegewar.utils.SiegeWarDistanceUtil;
+import com.palmergames.bukkit.towny.TownyAPI;
+import com.palmergames.bukkit.towny.TownyUniverse;
+import com.palmergames.bukkit.towny.event.actions.TownyDestroyEvent;
+import com.palmergames.bukkit.towny.exceptions.TownyException;
+import com.palmergames.bukkit.towny.object.Resident;
+import com.palmergames.bukkit.towny.object.Town;
+import net.md_5.bungee.api.ChatMessageType;
+import net.md_5.bungee.api.chat.TextComponent;
+import org.bukkit.*;
+import org.bukkit.block.Block;
+import org.bukkit.block.Container;
+
+/**
+ * This class handles siege-related destroy-block requests
+ *
+ * @author Goosius
+ */
+public class DestroyBlock {
+
+	/**
+	 * Evaluates a block destroy request.
+	 *
+	 * @param event The event object
+	 */
+	public static void evaluateSiegeWarDestroyBlockRequest(TownyDestroyEvent event) {    
+            //Ensure siege is enabled in this world
+            Block block = event.getBlock();
+			if (!TownyAPI.getInstance().getTownyWorld(block.getWorld()).isWarAllowed())
+				return;
+
+			//If the event has already been cancelled by Towny...
+			if(event.isCancelled()) {		
+				if(!SiegeWarSettings.isWallBreachingEnabled())
+					return; //Without wall breaching, SW doesn't un-cancel events
+				Town town = TownyAPI.getInstance().getTown(block.getLocation());
+				if(town == null)
+					return; //SW currently doesn't un-cancel wilderness events
+				if(!SiegeController.hasActiveSiege(town))
+					return; //SW doesn't un-cancel events in unsieged towns				
+
+				//Ensure player is on the town-hostile siege side				
+				Resident resident = TownyAPI.getInstance().getResident(event.getPlayer());
+				if(resident == null)
+					return;
+				if(!resident.hasNation())
+					return;
+				Siege siege = SiegeController.getSiege(town);
+				SiegeSide playerSiegeSide = SiegeWarAllegianceUtil.calculateCandidateSiegePlayerSide(event.getPlayer(), resident.getTownOrNull(), siege);
+				if(playerSiegeSide == SiegeSide.NOBODY)
+					return; 
+				switch(siege.getSiegeType()) {
+					case CONQUEST:
+					case SUPPRESSION:
+						if(playerSiegeSide != SiegeSide.ATTACKERS)
+							return;
+						break;
+					case REVOLT:
+					case LIBERATION:
+						if(playerSiegeSide != SiegeSide.DEFENDERS)
+							return;
+						break;
+				}
+
+				//Ensure player has permission
+				if (!TownyUniverse.getInstance().getPermissionSource().testPermission(event.getPlayer(), SiegeWarPermissionNodes.SIEGEWAR_NATION_SIEGE_USE_BREACH_POINTS.getNode())) {
+					event.setMessage(Translation.of("msg_err_action_disable"));
+					return; 
```

Comment:
- I see what you mean.
- But that would mean that regular soldiers trying to break blocks would just get normal Towny "you cannot break blocks" message, instead of the different "permission denied"
- I think that difference would help support teams to diagnose and fix issues.  But I don't mind. So, move/don't-move?

Reply:
If its a unique message then its fine to show a message, might still want to move it up higher so it gets handled earlier on.

Reply:
Will do

Reply:
Done